onnxruntime/core/framework/TensorSeq.h

```diff
@@ -56,16 +58,29 @@ class TensorSeq {
     return tensors_.cend();
   }
 
-  // Get by index
+  // Get onnxruntime::Tensor by index
   const Tensor& Get(size_t i) const {
+    return GetAt(i).Get<Tensor>();
+  }
+
+  // Get OrtValue by index
+  const OrtValue& GetAt(size_t i) const {
     ORT_ENFORCE(i < tensors_.size());
     return tensors_[i];
   }
 
+  void Add(const OrtValue& tensor) {
+    ORT_ENFORCE(IsSameDataType(tensor.Get<Tensor>()),
+                "TensorSeq: tensor to be added has a different data type.");
+    tensors_.push_back(tensor);
+  }
+
   void Add(Tensor&& tensor) {
     ORT_ENFORCE(IsSameDataType(tensor),
                 "TensorSeq: tensor to be added has a different data type.");
-    tensors_.push_back(std::move(tensor));
+    auto ml_tensor = TensorTypeBase::Type();
+    auto tensor_ptr = std::make_unique<Tensor>(std::move(tensor));
+    Add(OrtValue(tensor_ptr.release(), ml_tensor, ml_tensor->GetDeleteFunc()));
```

Comment:
Use Tensor::InitOrtValue() overloads to avoid code duplication.

Reply:
This is important, because we want to manage heap allocations in the same place due to RealTime requirements.

Reply:
The problem with the current `Tensor::InitOrtValue()` overloads is that they take raw data and creates a new tensor from it, but in this case we already have a `Tensor`. Calling `Tensor::InitOrtValue()` would require us to still manually call `tensor_ptr.release()` here, otherwise the raw data will get destroyed when this tensor goes out of scope.

Reply:
Please, add a new method overload, as I suggested in the original comment, so the copy construction and `std::make_unique` takes place inside the method just as any other relevant code.

Reply:
I take that back. I think this [overload](https://github.com/microsoft/onnxruntime/blob/main/include/onnxruntime/core/framework/tensor.h#L84) can be used to create a copy of the Tensor with its data.

Reply:
I added an overload.